app/services/katello/pulp3/api/ansible_collection.rb

```diff
@@ -25,7 +25,10 @@ def self.repository_sync_url_class
         end
 
         def api_client
-          PulpAnsibleClient::ApiClient.new(smart_proxy.pulp3_configuration(PulpAnsibleClient::Configuration))
+          client = PulpAnsibleClient::ApiClient.new(smart_proxy.pulp3_configuration(PulpAnsibleClient::Configuration))
+          request_id = ::Logging.mdc['request']
+          client.default_headers['Correlation-ID'] = request_id if request_id
+          client
```

Comment:
can we extract this out in some way so the code isn't repeated?

Reply:
one way potentially would be to refactor all these classes to have an 'api_client_class' method, and then api_client could live in the parent class  "Katello::Pulp3::Api::Core" and have the logic to do all this

Reply:
re extraction refactor:Absolutely. I considered it, but wanted to get eyes on the overall effort first.